google/resumable_media/_helpers.py

```diff
@@ -162,23 +162,43 @@ def wait_and_retry(func, get_status_code, retry_strategy):
     Returns:
         object: The return value of ``func``.
     """
-    response = func()
-    if get_status_code(response) not in RETRYABLE:
-        return response
-
     total_sleep = 0.0
     num_retries = 0
     base_wait = 0.5  # When doubled will give 1.0
-    while retry_strategy.retry_allowed(total_sleep, num_retries):
+
+    # Set the retriable_exception_type if possible. We expect requests to be
+    # present here and the transport to be using requests.exceptions errors,
+    # but due to loose coupling with the transport layer we can't guarantee it.
+    try:
+        import requests
+
+        retriable_exception_type = requests.exceptions.ConnectionError
```

Comment:
The `unit_2-2.7` and `system-2.7` tests are failing here, with 

```python
  File "/tmpfs/src/github/google-resumable-media-python/google/resumable_media/_helpers.py", line 175, in wait_and_retry
    retriable_exception_type = requests.exceptions.ConnectionError
AttributeError: 'module' object has no attribute 'exceptions'
```

Importing `requests.exceptions` explicitly would probably fix that.

Reply:
Whoops, thanks for catching that. I should be consistent about which of requests.ConnectionError and requests.exceptions.ConnectionError I use here (they refer to the same exception) to avoid these issues. I'll standardize on one or another and change imports to reflect.